### PR TITLE
Add Qwen-Image

### PR DESCRIPTION
**Pull request by atakaha**
Enable qwen-image with Gaudi specific optimization. Rewrite RoPE from complex to cos/sin.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from diffusers import QwenImagePipeline
+
+        >>> pipe = QwenImagePipeline.from_pretrained("Qwen/Qwen-Image", torch_dtype=torch.bfloat16)
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py]: Can you add HPU equivalent standalone example (e.g. with GaudiQwenImagePipeline)?

Comment (atakaha) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py]: @dsocek, thank you for pointing. The section is updated.

File: examples/stable-diffusion/README.md [hunk 2]
```diff
@@ -178,6 +178,32 @@ FLUX in quantization mode by setting runtime variable `QUANT_CONFIG=quantization
...
+    --scheduler flow_match_euler_discrete \
+    --use_habana \
+    --use_hpu_graphs \
+    --gaudi_config Habana/stable-diffusion \
+    --sdp_on_bf16 \
+    --bf16
+```
+
+> [!NOTE]
+> If you don't add `--negative_prompts` then empty string will added to it as default value.
```
Comment (regisss) [inline examples/stable-diffusion/README.md]:
```suggestion
> If you don't add `--negative_prompts` then empty string will be added to it as default value.
```

File: optimum/habana/diffusers/modeling_utils.py
Comment (regisss) [inline optimum/habana/diffusers/modeling_utils.py:1]: Could we do this monkey patching directly in the `__init__` of the `GaudiDiffusionPipeline`? https://github.com/huggingface/optimum-habana/blob/840e497b64810bc1813cf38ee24dd4590d55c55f/optimum/habana/diffusers/pipelines/pipeline_utils.py#L145

File: examples/stable-diffusion/README.md (hunk 2 above)
Comment (atakaha) [reply examples/stable-diffusion/README.md]: fix the wording

File: optimum/habana/diffusers/modeling_utils.py
Comment (atakaha) [reply optimum/habana/diffusers/modeling_utils.py:1]: I added like below, but didn't work.
`            from ..models.transformers import (
                GaudiQwenTimestepProjEmbeddings,
                GaudiQwenEmbedRope,
                GaudiQwenDoubleStreamAttnProcessor2_0
            )
            diffusers.models.transformers.transformer_qwenimage.QwenTimestepProjEmbeddings = GaudiQwenTimestepProjEmbeddings
            diffusers.models.transformers.transformer_qwenimage.QwenEmbedRope = GaudiQwenEmbedRope
            diffusers.models.transformers.transformer_qwenimage.QwenDoubleStreamAttnProcessor2_0 = GaudiQwenDoubleStreamAttnProcessor2_0`

Comment (atakaha) [reply optimum/habana/diffusers/modeling_utils.py:1]: monkey patch doesn't work in `__init_`_ but `from_pretrained()` of the `GaudiDiffusionPipeline`. Placed it in `from_pretrained()`.

Comment (regisss) [reply optimum/habana/diffusers/modeling_utils.py:1]: Sounds good, thanks for looking further into it!

File: optimum/habana/diffusers/pipelines/pipeline_utils.py [hunk 3]
```diff
@@ -375,6 +375,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         diffusers.pipelines.pipeline_utils.LOADABLE_CLASSES = GAUDI_LOADABLE_CLASSES
         diffusers.pipelines.pipeline_utils.ALL_IMPORTABLE_CLASSES = GAUDI_ALL_IMPORTABLE_CLASSES
 
+        from optimum.habana.diffusers.modeling_utils import adapt_qwenimage_transfomers_to_gaudi
+
+        adapt_qwenimage_transfomers_to_gaudi()
```
Comment (regisss) [inline optimum/habana/diffusers/pipelines/pipeline_utils.py]: Let's do the monkey patching explicitly here and not create a new `modeling_utils.py` file?

Comment (atakaha) [reply optimum/habana/diffusers/pipelines/pipeline_utils.py]: Placed extracted monkey patch at the place and removed `modeling_utils.py`.
